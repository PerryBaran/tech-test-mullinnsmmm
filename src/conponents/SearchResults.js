import React from "react";
import "../styles/searchresults.css";

const SearchResults = ({ results }) => {
    console.log(results);
    if(!results) {
        return <p></p>
    } else if (!results.length) {
        return <p className="results-message">No results</p>
    } else {
    return (
        <>
        <div className="search-results">
            {results.map((result) => (
                <img 
                key={result}
                className="card-image" 
                src={ result } 
                alt="spaceImage" />
            ))}
            </div>
        </>
        );
    }
};

export default SearchResults;