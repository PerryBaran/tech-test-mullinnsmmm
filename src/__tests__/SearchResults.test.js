import React from "react";
import { render } from "@testing-library/react";
import SearchResults from "../conponents/SearchResults";

describe("Search Results", () => {
    it("renders correctly", () => {
        const results = [];
        const { asFragment } = render(<SearchResults results={results}/>);
        expect(asFragment()).toMatchSnapshot();
    })
});