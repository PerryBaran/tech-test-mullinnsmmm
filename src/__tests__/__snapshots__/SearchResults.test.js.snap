// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results renders correctly 1`] = `
<DocumentFragment>
  <p
    class="results-message"
  >
    No results
  </p>
</DocumentFragment>
`;

exports[`Search Results returns image renders image 1`] = `
<DocumentFragment>
  <div
    class="searchResults"
  >
    <img
      alt="searchResults"
      class="search-images"
      src="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
    />
    <p>
      Search Result
    </p>
  </div>
</DocumentFragment>
`;
